main.go

```diff
@@ -0,0 +1,202 @@
+	logModeEnvVar = "ZAP_DEVELOPMENT_MODE"
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	//+kubebuilder:scaffold:scheme
+}
+
+func main() {
+	if err := startManager(); err != nil {
+		setupLog.Error(err, "unable to start manager")
+		os.Exit(1)
+	}
+}
+
+func startManager() error {
+	setupLog.Info("Starting k8s-service-discovery...")
+
+	options, err := getK8sManagerOptions()
+	if err != nil {
+		return fmt.Errorf("failed to get manager options: %w", err)
+	}
+
+	k8sManager, err := ctrl.NewManager(ctrl.GetConfigOrDie(), options)
+	if err != nil {
+		return fmt.Errorf("failed to create new manager: %w", err)
+	}
+
+	if err := createIngressClassComponent(k8sManager); err != nil {
+		return fmt.Errorf("failed to create ingress class creator: %w", err)
+	}
+
+	if err := configureManager(k8sManager, options.Namespace); err != nil {
+		return fmt.Errorf("failed to configure service discovery manager: %w", err)
+	}
+
+	if err := startK8sManager(k8sManager); err != nil {
+		return fmt.Errorf("failed to configure service discovery manager: %w", err)
+	}
+
+	return nil
+}
+
+func configureManager(k8sManager manager.Manager, namespace string) error {
+	if err := configureReconciler(k8sManager, namespace); err != nil {
+		return fmt.Errorf("failed to configure reconciler: %w", err)
+	}
+
+	// This kubebuilder marking inserts boilerplate code required for the manager. Do not remove it!
+	//+kubebuilder:scaffold:builder
+
+	if err := addChecks(k8sManager); err != nil {
+		return fmt.Errorf("failed to configure reconciler: %w", err)
+	}
+
+	return nil
+}
+
+func configureLogger() error {
+	logMode := false
+
+	logModeEnv, found := os.LookupEnv(logModeEnvVar)
+	if found {
+		parsedLogMode, err := strconv.ParseBool(logModeEnv)
+		if err != nil {
+			return fmt.Errorf("failed to parse %s; valid values are true or false: %w", logModeEnv, err)
+		}
+		logMode = parsedLogMode
+	}
+
+	opts := zap.Options{
+		Development: logMode,
+	}
+	opts.BindFlags(flag.CommandLine)
+	flag.Parse()
+
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+
+	return nil
+}
+
+func getK8sManagerOptions() (manager.Options, error) {
+	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
+	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
+		"Enable leader election for controller manager. "+
+			"Enabling this will ensure there is only one active controller manager.")
+
+	if err := configureLogger(); err != nil {
+		return manager.Options{}, fmt.Errorf("failed to configure logger: %w", err)
+	}
+
+	options := ctrl.Options{
+		Scheme:                 scheme,
+		MetricsBindAddress:     metricsAddr,
+		Port:                   9443,
+		HealthProbeBindAddress: probeAddr,
+		LeaderElection:         enableLeaderElection,
+		LeaderElectionID:       "92a787f2.cloudogu.com",
```

Comment:
I actually do not know. It was generated by the kubebuilder. I would prefer to not change things where I do not know how it will affect the application.